keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolver.kt

```diff
@@ -114,30 +117,34 @@ class ImageResolver(
     ) ?: throw NoImageFound(imageProvider.packageName)
 
     log.info("Image found for {}: {}", imageProvider.packageName, image)
-    return VersionedNamedImage(image, null, null)
+    return VersionedNamedImage(listOf(image), null, null)
   }
 
   private fun Resource<ClusterSpec>.withVirtualMachineImages(image: VersionedNamedImage): Resource<ClusterSpec> {
     val imageIdByRegion = image
-      .namedImage
-      .amis
+      .namedImages
+      .map { it.amis }
+      .reduceRight { acc, map -> acc + map }
```

Comment:
Woo, foldr!

Reply:
I'm getting rid of this complexity right now!